registry/storage/driver/s3/s3.go

```diff
@@ -209,7 +217,12 @@ func New(params DriverParameters) (*Driver, error) {
 	}
 
 	s3obj := s3.New(auth, params.Region)
-	bucket := s3obj.Bucket(params.Bucket)
+
+	if params.UserAgent != "" {
+		s3obj.Client = &http.Client{
+			Transport: transport.NewTransport(http.DefaultTransport, transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{params.UserAgent}})),
```

Comment:
Minor cosmetic point, but I think this would be easier to read if broken over multiple lines, like this:

	s3obj.Client = &http.Client{
		Transport: transport.NewTransport(http.DefaultTransport,
			transport.NewHeaderRequestModifier(http.Header{
				http.CanonicalHeaderKey("User-Agent"): []string{params.UserAgent},
			}),
		),
	}